src/Raven.Server/RavenServer.cs

```diff
@@ -355,12 +357,23 @@ private void ListenToNewTcpConnection(TcpListener listener)
                                 tcp.PinnedBuffer
                                 ))
                             {
-                                header = JsonDeserializationClient.TcpConnectionHeaderMessage(headerJson);
+                                header = NewClient.Client.Json.JsonDeserializationClient.TcpConnectionHeaderMessage(headerJson);
```

Comment:
Why is this the server calling the client deserialization? If this is used on both ends, better to move the method to the base class.